runtime/onert/api/src/nnfw_api_internal.cc

```diff
@@ -1154,7 +1141,7 @@ bool nnfw_session::isStateModelLoaded()
   if (_state == State::MODEL_LOADED)
   {
     assert(_subgraphs);
-    assert(_compiler);
+    assert(!_compiler);
```

Comment:
I think the following is more readable. `assert(ptr)` is acceptable, but `assert(!ptr)` is harder because it requires an additional operation (ptr to bool conversion &  boolean negation). I would like to create an PR for this.
```
assert(_compiler == nullptr);
```

Back to this PR:
It means `_compiler` should not be created after `MODEL_LOADED`. Why do we need to create compiler before compile? It also causes duplications for every load...() families.

Reply:
>`assert(ptr)` is acceptable, but `assert(!ptr)` is harder 

Agree